README.md

```diff
@@ -226,27 +226,60 @@ The VC++ linker supports `/SOURCELINK` [switch](https://docs.microsoft.com/en-us
+| No increase in size of assemblies       | ✅                          | ✅                      | ❌                |
+| Supported by all package feeds          | ❌ (Supported on NuGet.org) | ✅                      | ✅                |
+| Supports all symbol types               | ❌ (Portable symbols only)  | ✅                      | ✅                |
+
+### .snupkg symbol packages
+
 If you distribute the library via a package published to [NuGet.org](http://nuget.org), it is recommended to build a [symbol package](https://docs.microsoft.com/en-us/nuget/create-packages/symbol-packages-snupkg) and publish it to [NuGet.org](http://nuget.org) as well. This will make the symbols available on [NuGet.org symbol server](https://docs.microsoft.com/en-us/nuget/create-packages/symbol-packages-snupkg#nugetorg-symbol-server), where the debugger can download it from when needed.
 
-Alternatively, Portable PDBs can be included in the main NuGet package by setting the following property in your project:
+.snupkg symbol packages have following limitations:
+
+- They do not support Windows PDBs (generated by VC++, or for managed projects that set build property `DebugType` to `full`)
+- They require the library to be built by newer C#/VB compiler (Visual Studio 2017 Update 9).
+- The consumer of the package also needs Visual Studio 2017 Update 9 or newer.
+- Not supported by [Azure DevOps Artifacts](https://azure.microsoft.com/en-us/services/devops/artifacts) service.
+
+If a .snupkg does not work for your scenario, consider including debug information in the main package via one of the alternatives.
+
+### Include in main package
+
+You can include PDB files in the main NuGet package by setting the following property in your project:
 
 ```xml
 <PropertyGroup>
   <AllowedOutputExtensionsInPackageBuildOutputFolder>$(AllowedOutputExtensionsInPackageBuildOutputFolder);.pdb</AllowedOutputExtensionsInPackageBuildOutputFolder>
 </PropertyGroup>
 ```
 
-Keep in mind that including PDBs in the .nupkg increases the size of the package and thus restore time for projects that consume your package, regardless of whether the user needs to debug through the source code of your library or not.
+> [!IMPORTANT]
+> Keep in mind that including debug information in the .nupkg increases the size of the package and thus restore time for projects that consume your package, regardless of whether the user needs to debug the source code of your library or not.
 
-.snupkg symbol packages have following limitations:
+> [!IMPORTANT]
+> When including PDB files in the main package, projects that consume the package must _also_ opt-in to copying the symbols into their own output directory. Starting in .NET 7 this can be controlled via the [`CopyDebugSymbolFilesFromPackages`](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#copydebugsymbolfilesfrompackages) property.
```

Comment:
@baronfel Shouldn't this property be true by default?

Reply:
https://github.com/dotnet/sdk/pull/27580 - it was added in this PR and no default was set (so effectively `false`) because it would bloat application sizes. We could potentially make another push here, but we didn't feel that there was enough justification at the time.

Reply:
I believe I've addressed your concerns, so I'm marking this thread as Resolved. If you disagree or have additional questions, please reactivate and let me know. Thanks!

Reply:
@baronfel Not sure how does it bloat app size. Does publishing automatically include PDBs that are in the bin directory? I would hope not.

Reply:
Yes, all of the `ReferenceCopyLocalPaths` are copied to publish output by default: https://github.com/dotnet/sdk/blob/2c86f58b098147e268e4a19ce52294128f0b4b2a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets#L683-L695

Reply:
Is that expected? I'd not expect PDBs getting published in release build.
If someone sets `CopyDebugSymbolFilesFromPackages` to `true`, how do they avoid publishing PDBs in the app?

Reply:
The same way you prevent anything being copied - setting the `Private="true"` metadata on the `Content` items. But yes, this is a fun area overall, and the interaction of these Items is non-intuitive.

Reply:
@tmat, anything else here? Is this OK to merge?